### PR TITLE
Fix the dllimport code generation for opaque type

### PR DESCRIPTION
**Pull request by kaadam**
Fixes: #1407

### CODE REVIEW

**Review by pawosm-arm — APPROVED** [review]

**Comment by bryanpkc** [comment]
@kaadam Can you add a test case for this under `test/llvm_ir_correct/`?

**Comment by kaadam** [comment]
> @kaadam Can you add a test case for this under `test/llvm_ir_correct/`?

@bryanpkc Yes, I do a test case for this. My plan is to cover this one as well: #1177

**Review by shivaramaarao — APPROVED** [review]

**Comment by bryanpkc** [comment]
> > @kaadam Can you add a test case for this under `test/llvm_ir_correct/`?
> 
> @bryanpkc Yes, I do a test case for this. My plan is to cover this one as well: #1177

@kaadam Sorry, I am not sure whether you were intending to add that test case to this PR or not. If you have one, could you please add it to this PR so we can merge it?

**Comment by bryanpkc** [comment]
@kaadam Gentle ping.

**Comment by kaadam** [comment]
> @kaadam Gentle ping.

@bryanpkc Sorry, for the delayed response. I was out of office. I will add the tests to this PR.
I'm not sure about what the community prefer: Shall I create a minimized tests for that (with dummy external module)? Or could I reuse the existing ieee_arithmetic or ieee_exceptions modules in the test?

**Comment by bryanpkc** [comment]
@kaadam Sorry for my late reply. A minimized test case would be better IMO.

**Comment by bryanpkc** [comment]
@kaadam Any chance you could add a test to this PR and land it?

**Comment by kaadam** [comment]
> @kaadam Any chance you could add a test to this PR and land it?

@bryanpkc Sure, I'm on it to create a test case for this. Sorry, for the delay.

File: test/llvm_ir_correct/dllimport_test.f90_mod.f90 [hunk 1]
```diff
@@ -0,0 +1,28 @@
+!
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! REQUIRES: system-windows
+! RUN: true
```
Comment (pawosm-arm) [inline test/llvm_ir_correct/dllimport_test.f90_mod.f90:7]: is anything being tested here?

Comment (pawosm-arm) [reply test/llvm_ir_correct/dllimport_test.f90_mod.f90:7]: ok, I suppose `%flang -S -emit-llvm -S -emit-llvm %s_mod.f90 %s` these things relate to each other. Would be great to comment that - one can't figure it out by looking at the file itself.

Comment (kaadam) [reply test/llvm_ir_correct/dllimport_test.f90_mod.f90:12]: Added comments to tests. Thanks.

**Review by bryanpkc — APPROVED** [review]

**Pull request merged**